Core/Core/Store/Store.swift

```diff
@@ -104,11 +104,15 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
     }
 
     public subscript(indexPath: IndexPath) -> U.Model? {
-        return frc.object(at: indexPath)
+        let object = frc.object(at: indexPath)
+        if object.isDeleted {
+            return nil
+        }
+        return object
     }
 
     public subscript(index: Int) -> U.Model? {
-        return frc.object(at: IndexPath(row: index, section: 0))
+        return self[IndexPath(row: index, section: 0)]
     }
 
     public func numberOfObjects(inSection section: Int) -> Int {
```

Comment:
does this method still work?

Reply:
would this still cause a tableview to have say x number of rows but if one is nil, then that count would be off by 1?

Reply:
Yeah, the counts are all still the same. The FRC is including deleted objects in these counts during certain race conditions. So the object is there, it's just that all its properties are faults and `.isDeleted` is true.

The problem is that when we try to access non-optional properties that are faults we get a crash.

I guess this is one reason we _should_ make all NSManaged properties optional but yuck.

Reply:
> would this still cause a tableview to have say x number of rows but if one is nil, then that count would be off by 1?

No, it shouldn't. The count methods are including these deleted objects.

There is an object at the indexPath and the count includes it, but the object has been deleted. So now when we go asking for the object we will get `nil` back so that we don't try to access properties on a deleted object.

This is where the most prevalent crash is at right now:
https://github.com/instructure/canvas-ios/blob/60296d7437690aa21bb4151e21ddd66d426384e4/Core/Core/Pages/PageList/PageListViewController.swift#L167-L174

On line 172 we pass a deleted page to `cell.update`. In that update method we access non-optional properties on the deleted page. That causes a crash because the properties are nil because the page has been deleted.

Reply:
I can update the test to verify the `numberOfSections` is still correct.